### PR TITLE
chore: upgrade compose to fix NavBackStackEntry crashes [RC] [WPB-2321]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-2321" title="WPB-2321" target="_blank">BugWPB-2321</a>  [Android][Naughty Narwhal+] You cannot access the NavBackStackEntry's ViewModels after the NavBackStackEntry is destroyed.
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Copy of https://github.com/wireapp/wire-android-reloaded/pull/1965

### Issues

Some users still have the infamous NavBackStackEntry crashes.

### Causes (Optional)

In some places, we inject view model (using `hiltViewModel()` for instance) inside a composable that uses `SubcomposeLayout` which means the injection happens inside the subcomposition. When the parent `NavBackStackEntry` is being destroyed, the subcomposition still can happen but it will crash because this destroyed entry is the `ViewModelStoreOwner`.

### Solutions

Google provided a fix in a new version of compose, more info here: https://issuetracker.google.com/issues/254645321
With the new compose, unfortunately `backQueue` is private, the only way to get the stack is by using internal `currentBackStack`.

#### How to Test

Try to navigate to other user profile and back quickly so that subcomposition isn't yet executed, or just do something like this:
```
LaunchedEffect(Unit) {
    delay(50L)
    eventsHandler.navigateBack()
}
```

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by alexandreferris — APPROVED** [review]

**Review by Garzas — APPROVED** [review]

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -83,12 +83,13 @@ object Libraries {
         const val desugaring = "1.1.5"
         const val workManager = "2.8.1"
         const val fragment = "1.5.6"
-        const val compose = "1.4.1"
+        const val compose = "1.5.0-beta03"
```
Comment (vitorhugods) [inline buildSrc/src/main/kotlin/Dependencies.kt:86]: Are we releasing a new hotfix?

`dependencies.kt` looks very different on `develop` now.

Comment (saleniuk) [reply buildSrc/src/main/kotlin/Dependencies.kt:86]: Yes, this PR is considered a fix for all remaining NavBackStackEntry crashes, I just wanted to upgrade these libraries that are necessary to get rid of these crashes. @vitorhugods do you think we shouldn't update RC with these changes at this point?

The PR for develop was raised earlier, it's here: https://github.com/wireapp/wire-android-reloaded/pull/1965

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1971/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.4(7079534)(7079534)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.4(7079534)(7079534)-staging-compat.apk)

**Comment by tmspzz** [comment]
Yes, let's not change that for RC.

**Pull request merged**